addons/binding/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/discovery/SqueezeBoxServerDiscoveryService.java

```diff
@@ -0,0 +1,275 @@
+				NetworkInterface networkInterface = (NetworkInterface) interfaces
+						.nextElement();
+				if (networkInterface.isLoopback() || !networkInterface.isUp()) {
+					continue;
+				}
+				for (InterfaceAddress interfaceAddress : networkInterface
+						.getInterfaceAddresses()) {
+					InetAddress[] broadcast = new InetAddress[3];
+					broadcast[0] = InetAddress.getByName("224.0.0.1");
+					broadcast[1] = InetAddress.getByName("255.255.255.255");
+					broadcast[2] = interfaceAddress.getBroadcast();
+					for (InetAddress bc : broadcast) {
+						if (bc != null) {
+							try {
+								logger.trace("Broadcasting {}", sendData);
+								DatagramPacket sendPacket = new DatagramPacket(
+										sendData, sendData.length, bc,
+										DISCO_PORT);
+								socket.send(sendPacket);
+							} catch (IOException e) {
+								logger.debug(
+										"IO error during SqueezeBoxServer discovery: {}",
+										e.getMessage());
+							} catch (Exception e) {
+								logger.debug(e.getMessage(), e);
+							}
+							logger.trace(
+									"Request packet sent to: {} Interface: {}",
+									bc.getHostAddress(),
+									networkInterface.getDisplayName());
+						}
+					}
+				}
+			}
+			logger.trace("Done looping over all network interfaces. Now waiting for a reply!");
+
+		} catch (IOException e) {
+			logger.debug("IO error during SqueezeBoxServer discovery: {}",
+					e.getMessage());
+		}
+	}
+
+	/**
+	 * Based off the MaxCube discovery code, thanks! Listens for replies to our
+	 * discovery message
+	 */
+	private void receiveDiscoveryMessage() {
+		try {
+			while (discoveryRunning) {
+				// Wait for a response
+				byte[] recvBuf = new byte[1500];
+				DatagramPacket receivePacket = new DatagramPacket(recvBuf,
+						recvBuf.length);
+				logger.trace("Waiting for packets");
+				socket.receive(receivePacket);
+
+				String message = new String(receivePacket.getData()).trim();
+				logger.trace("Broadcast response from {} : {} '{}'",
+						receivePacket.getAddress(), message.length(), message);
+				Matcher matcher = replyPattern.matcher(message);
+				if (matcher.matches()) {
+					// we have a packet
+					String ip = receivePacket.getAddress().getHostAddress();
+					String name = matcher.group(1).trim();
+					String webPort = matcher.group(2).trim();
+					String uuid = matcher.group(4).trim().toUpperCase();
+
+					logger.debug("Found server {} at {}:{} with uuid {}", name,
+							ip, webPort, uuid);
+
+					try {
+						int cliPort = HttpUtils.getCliPort(ip,
+								Integer.parseInt(webPort));
+
+						Map<String, Object> properties = new HashMap<>(2);
+						properties.put("ipAddress", ip);
+						properties.put("webport", new Integer(webPort));
+						properties.put("cliPort", new Integer(cliPort));
+
+						ThingUID uid = new ThingUID(
+								SqueezeBoxBindingConstants.SQUEEZEBOXSERVER_THING_TYPE,
+								uuid);
+
+						DiscoveryResult result = DiscoveryResultBuilder
+								.create(uid)
```

Comment:
So I have a better solution, how about I just remove this whole class ;-) After testing some more, UPNP seems to work fine in all cases. This method must have been a precursor to using a standard discovery protocol and seems to work fine with the three instances of a squeeze server i'm testing with. The documentation on this stuff is little to none unfortunately (unless you want to read 15 year old perl code that the server is written in)